tests/checker/good/gold/dead_code_test10.gold

```diff
@@ -0,0 +1,84 @@
+{
+  "cashflow_tags": { "State variables": [], "ADT constructors": [] },
+  "contract_info": {
+    "scilla_major_version": "0",
+    "vname": "Dead",
+    "params": [],
+    "fields": [],
+    "transitions": [ { "vname": "Foo", "params": [] } ],
+    "procedures": [
+      {
+        "vname": "Bar",
+        "params": [ { "vname": "a", "type": "dead_code_test11.A" } ]
+      }
+    ],
+    "events": [],
+    "ADTs": [
+      {
+        "tname": "dead_code_test11.A",
+        "tparams": [],
+        "tmap": [
+          { "cname": "dead_code_test11.A1", "argtypes": [] },
+          { "cname": "dead_code_test11.A2", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Option",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Some", "argtypes": [ "'A" ] },
+          { "cname": "None", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Bool",
+        "tparams": [],
+        "tmap": [
+          { "cname": "True", "argtypes": [] },
+          { "cname": "False", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Nat",
+        "tparams": [],
+        "tmap": [
+          { "cname": "Zero", "argtypes": [] },
+          { "cname": "Succ", "argtypes": [ "Nat" ] }
+        ]
+      },
+      {
+        "tname": "List",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Cons", "argtypes": [ "'A", "List ('A)" ] },
+          { "cname": "Nil", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Pair",
+        "tparams": [ "'A", "'B" ],
+        "tmap": [ { "cname": "Pair", "argtypes": [ "'A", "'B" ] } ]
+      }
+    ]
+  },
+  "warnings": [
+    {
+      "warning_message": "Unused ADT constructor: A2",
+      "start_location": {
+        "file": "tests/contracts/dead_code_test11.scilla",
```

Comment:
If this is `tests/contracts/dead_code_test11.scilla` then no warnings on unused ADT constructors should be reported because of `transition Bar(x : A)`